lib/Alchemy/Phrasea/Core/Event/Subscriber/ElasticsearchSubscriber.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2014 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Alchemy\Phrasea\Core\Event\Subscriber;
+
+use Alchemy\Phrasea\Core\Event\CollectionEvent\CollectionIndexEvent;
+use Alchemy\Phrasea\Core\Event\DataboxEvent\DataboxIndexEvent;
+use Alchemy\Phrasea\Core\Event\DataboxEvent\IndexEvent;
+use Alchemy\Phrasea\Core\Event\RecordEvent\RecordIndexEvent;
+use Alchemy\Phrasea\Core\PhraseaEvents;
+use Alchemy\Phrasea\SearchEngine\Elastic\BulkOperation;
+use Alchemy\Phrasea\SearchEngine\Elastic\Indexer\RecordIndexer;
+use Alchemy\Phrasea\SearchEngine\Elastic\Indexer\TermIndexer;
+use Symfony\Component\EventDispatcher\Event as SfEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+use Elasticsearch\Client;
+
+class ElasticSearchSubscriber implements EventSubscriberInterface
+{
+    private $recordIndexer;
+    private $termIndexer;
+    private $client;
+    private $indexName;
+
+    /**
+     * keep tracks if index order has already been done
+     */
+    private $indexed = [];
+
+    public function __construct(RecordIndexer $recordIndexer, TermIndexer $termIndexer, Client $client, $indexName)
+    {
+        $this->recordIndexer = $recordIndexer;
+        $this->termIndexer = $termIndexer;
+        $this->client = $client;
+        $this->indexName = $indexName;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            PhraseaEvents::INDEX_DATABOX => ['onIndexDatabox', -12],
+            PhraseaEvents::INDEX_COLLECTION => ['onIndexCollection', -24],
+            PhraseaEvents::INDEX_RECORD => ['onIndexRecord', -48],
+        ];
+    }
+
+    public function onIndexDatabox(DataboxIndexEvent $event)
+    {
+        $databox = $event->getDatabox();
+
+        if ($this->isIndexationOrdered($event, $databox)) {
```

Comment:
This is kind of broken because only the first indexation process will be taken into account. For example if we have this use case
```php
$record->setStatus($status);
 // triggers indexations
$app['dispatcher']->dispatch(PhraseaEvents::RECORD_CHANGE_STATUS, new ChangeStatusEvent($record));

$record->setMetadata($metadata);
// triggers indexation
$app['dispatcher']->dispatch(PhraseaEvents::RECORD_CHANGE_METADATA, new ChangeMetadataEvent($record));
```

Only the last indexation order process must be fired.